src/StripeTests/Functional/TelemetryTest.cs

```diff
@@ -154,8 +191,36 @@ public void NoTelemetryWhenDisabled()
             var deserialized = JToken.Parse(payload);
             var requestId = (string)deserialized["last_request_metrics"]["request_id"];
             var duration = (long)deserialized["last_request_metrics"]["request_duration_ms"];
+            var usageRaw = deserialized["last_request_metrics"]["usage"];
+
+            List<string> usage = null;
+            if (usageRaw != null)
+            {
+                usage = usageRaw.Select(x => (string)x).ToList();
+            }
 
-            return requestIdMatcher(requestId) && durationMatcher(duration);
+            return requestIdMatcher(requestId) && durationMatcher(duration) && usageMatcher(usage);
+        }
+
+        private class TestEntity : StripeEntity<TestEntity>
+        {
+        }
+
+        private class TestService : Service<TestEntity>
+        {
+            public TestService(IStripeClient client)
+                : base(client)
+            {
+            }
+
+            public override string BasePath => "/v1/test";
+
+            public virtual void MakeRequestWithUsage(RequestOptions requestOptions)
+            {
+                RequestOptions ro = requestOptions.Clone();
+                ro.Usage = new List<string> { "llama", "bufo" };
```

Comment:
for the actual implementation we should try and avoid creating new lists and throwing them away for every call. Maybe a static field somewhere?

Reply:
Sounds good, I made the test reflect this to make it more likely to remember, but I'll bear in mind, too.